app/lib/clients/create_employees.rb

```diff
@@ -0,0 +1,54 @@
+class Clients::CreateEmployees
+  def initialize(employees, time_off)
+    @employees = employees
+    @time_off = time_off
+    @dates = create_dates
+  end
+
+  def create_dates
+    day_before_yesterday = (Time.zone.yesterday - 1).to_formatted_s(:iso8601)
+    yesterday = Time.zone.yesterday.to_formatted_s(:iso8601)
+    today = Time.zone.now.to_formatted_s(:iso8601)
+    tomorrow = Time.zone.tomorrow.to_formatted_s(:iso8601)
+    day_afer_tomorrow = (Time.zone.tomorrow + 1).to_formatted_s(:iso8601)
+    [day_before_yesterday, yesterday, today, tomorrow, day_afer_tomorrow]
+  end
+
+  def find_birthday_employees
+    birthdays = []
+    @employees.each do |employee|
+      if employee['birthday'].present? &&
+         @dates.any? { |d|
+           d.include? employee['birthday']
+         }
+        birthdays.append(employee)
+      end
+    end
+    birthdays
+  end
+
+  def find_anniversary_employees
+    anniversaries = []
+    @employees.each do |employee|
+      if employee['hireDate'].present? &&
+         @dates.any? { |d|
+           d.include? employee['hireDate']
+         }
+        anniversaries.append(employee)
+      end
+    end
+    anniversaries
+  end
+
+  def find_time_off_employees
+    whos_out = []
+    @employees.each do |employee|
+      if @time_off.any? { |e|
+        e['employeeId'].to_s.include? employee['id']
+      }
```

Comment:
What is the difference between the employee object and the e object? Would it be possible to only iterate over @time_off here?

Reply:
The e object refers to the employees that are in the time off array, and the employee object is one of all of the employees. I think we would need to iterate through all of the employees in the company as well as the ones in the time off array in order to find a match between the two groups. Do you have an example of what you mean?

Reply:
I was thinking perhaps performance could be improved by putting @employees data into a separate Map data structure for constant time lookup of @employees, and then you would only have to iterate over @time_off. I'm realizing now though that it seems like `employeeId` and `id` are not the same value because you use .include? to compare them?